dashboard/permissions_test.py

```diff
@@ -0,0 +1,96 @@
+"""
+Tests for the dashboard permissions
+"""
+from unittest.mock import Mock
+from django.http import Http404
+from django.db.models.signals import post_save
+from factory.django import mute_signals
+import ddt
+
+from courses.factories import ProgramFactory
+from micromasters.factories import UserFactory
+from dashboard.models import ProgramEnrollment
+from dashboard.permissions import CanReadIfStaffOrSelf
+from search.base import MockedESTestCase
+from roles.models import Role
+from roles.roles import (
+    Instructor,
+    Staff,
+)
+
+
+@ddt.ddt
+class CanReadIfStaffOrSelfTests(MockedESTestCase):
+    """
+    Tests
+    """
+
+    def test_anonymous_users_blocked(self):
+        """
+        Test that anonymous users get a 404
+        """
+        perm = CanReadIfStaffOrSelf()
+        request = Mock(user=Mock(is_anonymous=Mock(return_value=True)))
+        view = Mock(kwargs={'user': 'username'})
+        with self.assertRaises(Http404):
+            perm.has_permission(request, view)
+
+    def test_raise_if_requested_record_doesnt_exist(self):
+        """
+        Test that requests a nonexistent user gives a  404
+        """
+        perm = CanReadIfStaffOrSelf()
+        with mute_signals(post_save):
+            user = UserFactory.create()
+            request = Mock(user=user)
+            view = Mock(kwargs={'username': 'AFSDFASDFASDF'})
+            with self.assertRaises(Http404):
+                perm.has_permission(request, view)
+
+    def test_learner_can_get_own_dashboard(self):
+        """
+        Test that a user can get their own dashboard
+        """
+        perm = CanReadIfStaffOrSelf()
+        with mute_signals(post_save):
+            user = UserFactory.create()
+            request = Mock(user=user)
+            view = Mock(kwargs={'username': user.username})
+            assert perm.has_permission(request, view) is True
+
+    def test_learners_cannot_get_other_learners(self):
+        """
+        Normal users shouldn't be able to read each other
+        """
+        perm = CanReadIfStaffOrSelf()
+        with mute_signals(post_save):
+            user = UserFactory.create()
+            another_user = UserFactory.create()
+            request = Mock(user=user)
+            view = Mock(kwargs={'username': another_user.username})
+            with self.assertRaises(Http404):
+                perm.has_permission(request, view)
+
+    @ddt.data((Instructor), (Staff))
```

Comment:
huh, before I had to add those because of some error, but now it's not happening! lol